src/components/fields/location-input-group/location-modal/location-search/location-search.tsx

```diff
@@ -0,0 +1,749 @@
+import LocationPinBlack from "../../../../../assets/img/icons/location-pin-black.svg";
+import SearchSvg from "../../../../../assets/img/icons/search.svg";
+import { InfiniteScrollList, Prompt } from "../../../../elements";
+import { ILocationDisplayListParams, ILocationInputValues, TSinglePanelInputMode } from "../../types";
+import {
+	ButtonItem,
+	ButtonWrapper,
+	NoResultTitle,
+	ResultItem,
+	ResultItemPin,
+	ResultTitle,
+	ResultWrapper,
+	SearchBarContainer,
+	SearchBarCross,
+	SearchBarIcon,
+	SearchBarIconButton,
+	SearchBarInput,
+	SearchBarModalCross,
+	SearchWrapper,
+} from "./location-search.styles";
+
+interface IListItem extends ILocationInputValues {
+	displayText?: string;
+}
+
+export interface ILocationSearchProps {
+	id?: string;
+	onCancel: () => void;
+	onConfirm: () => void;
+	isSinglePanel: boolean;
+	singelPanelInputMode: TSinglePanelInputMode;
+	addressFieldPlaceholder?: string;
+	gettingCurrentLocation: boolean;
+	gettingCurrentLocationFetchMessage?: string;
+	locationListTitle?: string;
+	selectedAddressInfo: ILocationInputValues;
+	changeSelectedAddressInfo: (addressInfo: ILocationInputValues) => void;
+	pageSize?: number;
+	onOneMapError: () => void;
+	mustHavePostalCode?: boolean;
+	disableErrorPromptOnApp?: boolean;
+	reverseGeoCodeEndpoint?: string;
+	onGetLocationCallback: (lat?: number, lng?: number) => void;
+	onGetLocationError: (
+		geolocationPositionError?: GeolocationPositionError,
+		disableErrorPromptOnApp?: boolean
+	) => void;
+	onAddressChange: () => void;
+	showLocationModal: boolean;
+	mapPickedLatLng?: ILocationCoord;
+	didUserClickMap: boolean;
+	onClearInput: () => void;
+	formValues?: ILocationInputValues;
+	updateFormValues: (values: ILocationInputValues) => void;
+}
+
+const pagination = <T,>(array: T[], pageSize: number, pageNum: number) => {
+	return array.slice((pageNum - 1) * pageSize, pageNum * pageSize);
+};
+
+const fetchAddress = async (
+	query: string,
+	pageNum: number,
+	onSuccess?: (results: OneMapSearchResults) => void,
+	onFail?: (error: unknown) => void
+) => {
+	if (query) {
+		try {
+			const results = await OneMapService.searchByAddress({
+				searchValue: query,
+				getAddressDetails: OneMapBoolean.YES,
+				returnGeom: OneMapBoolean.YES,
+				pageNum: pageNum,
+			});
+			onSuccess?.(results);
+		} catch (error) {
+			onFail?.(error);
+		}
+	}
+};
+
+const debounceFetchAddress = debounce(fetchAddress, 500);
+
+const boldResultsWithQuery = (arr: OneMapSearchBuildingResult[], query: string) => {
+	const regex = new RegExp(query, "gi");
+	return arr.map((obj) => {
+		const newAddress = (obj.DISPLAY_ADDRESS || obj.ADDRESS).replace(regex, `<span class="keyword">${query}</span>`);
+		return {
+			...obj,
+			DISPLAY_ADDRESS: newAddress,
+		};
+	});
+};
+
+const cleanHtml = (html: string | undefined) => {
+	if (!html) return;
+	return {
+		__html: sanitizeHtml(html, {
+			allowedTags: ["span"],
+			allowedAttributes: {
+				span: ["class"],
+			},
+		}),
+	};
+};
```

Comment:
the codes are getting quite long so you can move these functions into a new `helpers.ts`, this file can sit outside the `location-modal` folder, in `location-field` folder to be shared with other components as well